src/mlpack/methods/ann/init_rules/glorot_init.hpp

```diff
@@ -77,8 +78,9 @@ class GlorotInitializationType
    *
    * @param W Weight matrix to initialize.
    */
-  template<typename eT>
-  void Initialize(arma::Mat<eT>& W);
+  template<typename MatType,
+      typename std::enable_if_t<IsMat<MatType>::value, bool> = false>
```

Comment:
```suggestion
      typename = typename std::enable_if_t<IsMat<MatType>::value, bool> = false>
```

I think it has to be written like this to compile successfully for all compilers; correct me if I'm wrong on that.